concrete-core/src/specification/engines/glwe_ciphertext_vector_encryption.rs

```diff
@@ -0,0 +1,52 @@
+use super::engine_error;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::{
+    GlweCiphertextVectorEntity, GlweSecretKeyEntity, PlaintextVectorEntity,
+};
+use concrete_commons::dispersion::Variance;
+
+engine_error! {
+    GlweCiphertextVectorEncryptionError for GlweCiphertextVectorEncryptionEngine @
+    PlaintextCountMismatch => "The key polynomial size must divide the plaintext count of the input \
+                               vector."
+}
+
+/// A trait for engines encrypting glwe ciphertext vectors.
+///
+/// # Semantics
+///
+/// This [pure](super#operation-semantics) operation generates a glwe ciphertext vector containing
+/// the piece-wise encryptions of the `input` plaintext vector, under the `key` secret key.
+///
+/// # Formal Definition
+pub trait GlweCiphertextVectorEncryptionEngine<SecretKey, PlaintextVector, CiphertextVector>:
+    AbstractEngine
+where
+    SecretKey: GlweSecretKeyEntity,
+    PlaintextVector: PlaintextVectorEntity<Representation = SecretKey::Representation>,
+    CiphertextVector: GlweCiphertextVectorEntity<
+        Representation = SecretKey::Representation,
+        KeyFlavor = SecretKey::KeyFlavor,
+    >,
+{
+    /// Encrypts a glwe ciphertext vector.
+    fn encrypt_glwe_ciphertext_vector(
+        &mut self,
+        key: &SecretKey,
+        input: &PlaintextVector,
+        noise: Variance,
```

Comment:
see https://github.com/zama-ai/concrete_internal/issues/177

Reply:
where do we use the generic dispersion then? NPE? maybe crypto API as well since it uses the NPE?

Reply:
I would say that we use them in human-facing APIs, like `concrete-boolean`  or  `concrete-crypto`. Basically anywhere it would bother a human to input a value as a `Variance`  directly.